FinSearchUnified/Tests/PluginTest.php

```diff
@@ -190,37 +191,74 @@ public function testProductIdExport($productId, $errorMessageOrProductCount)
 
     public function crossSellingCategoryProvider()
     {
+        $baseShop = Shopware()->Shop();
+        $subShop = Utility::createSubShop($baseShop);
+
         return [
             'No cross-sell categories configured' => [
                 'crossSellingCategories' => [],
-                'expectedCount' => 1
+                'expectedCount' => 1,
+                'baseShop' => $baseShop,
             ],
             'Article does not exist in cross-sell category configured' => [
                 'crossSellingCategories' => ['Deutsch>Genusswelten'],
-                'expectedCount' => 1
+                'expectedCount' => 1,
+                'baseShop' => $baseShop,
             ],
             'Article exists in one of the cross-sell categories configured' => [
                 'crossSellingCategories' => ['Deutsch>Genusswelten', 'Deutsch>Wohnwelten'],
-                'expectedCount' => 0
+                'expectedCount' => 0,
+                'baseShop' => $baseShop,
             ],
             'Article exists in all of cross-sell categories configured' => [
                 'crossSellingCategories' => ['Deutsch>Genusswelten', 'Deutsch>Wohnwelten', 'Deutsch>Beispiele'],
-                'expectedCount' => 0
+                'expectedCount' => 0,
+                'baseShop' => $baseShop,
+            ],
+            'No cross-sell categories for sub shop' => [
+                'crossSellingCategories' => ['Deutsch>Genusswelten', 'Deutsch>Wohnwelten', 'Deutsch>Beispiele'],
+                'expectedCount' => 1,
+                'baseShop' => $baseShop,
+                'subShop' => $subShop,
+                'subCrossSellingCategories' => [],
+            ],
+            'With cross-sell categories for sub shop' => [
+                'crossSellingCategories' => ['Deutsch>Genusswelten', 'Deutsch>Wohnwelten', 'Deutsch>Beispiele'],
+                'expectedCount' => 0,
+                'baseShop' => $baseShop,
+                'subShop' => $subShop,
+                'subCrossSellingCategories' => ['Deutsch>Beispiele'],
             ],
         ];
     }
 
     /**
      * @dataProvider crossSellingCategoryProvider
      *
-     * @param int[] $crossSellingCategories
+     * @param int[] $baseCrossSellingCategories
      * @param int $expectedCount
+     * @param Shop $baseShop,
+     * @param ?Shop $subShop,
+     * @param int[] $subCrossSellingCategories,
      */
-    public function testArticleExportWithCrossSellingCategories($crossSellingCategories, $expectedCount)
-    {
+    public function testArticleExportWithCrossSellingCategories(
```

Comment:
Does this fail if you remove `scope="shop"` from `FinSearchUnified/Resources/config/config_template.xml`? What's the error message?

Reply:
It does not seem to fail. As we need to set the config manually, I am not sure if we can check that.

Reply:
@dblaichinger and I have discussed this and we will leave this as it is, even though it does not explicitly test that the `scope` setting is checked, but we also couldn't really find any better alternatives.